src/Stackctl/Config/RequiredVersion.hs

```diff
@@ -0,0 +1,75 @@
+module Stackctl.Config.RequiredVersion
+  ( RequiredVersion(..)
+  , requiredVersionFromText
+  , isRequiredVersionSatisfied
+  ) where
+
+import Stackctl.Prelude
+
+import Data.Aeson
+import Data.List (uncons)
+import qualified Data.List.NonEmpty as NE
+import qualified Data.Text as T
+import Data.Version hiding (parseVersion)
+import qualified Data.Version as Version
+import Text.ParserCombinators.ReadP (readP_to_S)
+
+data RequiredVersion = RequiredVersion
+  { requiredVersionOp :: Text
+  , requiredVersionCompare :: Version -> Version -> Bool
+  , requiredVersionCompareWith :: Version
+  }
+
+instance Show RequiredVersion where
+  show RequiredVersion {..} =
+    unpack requiredVersionOp <> " " <> showVersion requiredVersionCompareWith
+
+instance FromJSON RequiredVersion where
+  parseJSON =
+    withText "RequiredVersion" $ either fail pure . requiredVersionFromText
+
+requiredVersionFromText :: Text -> Either String RequiredVersion
+requiredVersionFromText = fromWords . T.words
+ where
+  fromWords :: [Text] -> Either String RequiredVersion
+  fromWords = \case
+    [w] -> parseRequiredVersion "=" w
+    [op, w] -> parseRequiredVersion op w
+    _ -> Left "TODO"
```

Comment:
TODO

Reply:
Are you looking to merge this before fixing this?

Reply:
No, I plan to fix it before merging. Just didn't think it mattered for review.